cdap-api/src/main/java/co/cask/cdap/api/security/securestore/SecureStoreMetadata.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.security.securestore;
+
+import java.util.Date;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents the metadata for the data stored in the Secure Store.
+ */
+public final class SecureStoreMetadata {
+
+  private static final String DESCRIPTION_FIELD = "description";
```

Comment:
It's not obvious to the user that the property "description" is being used for extracting the description. I think it's better for the SecureStoreManager API to have "description" passed in. Or the other way is that we don't treat "description" differently but just leave it as a property.